src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -97,16 +98,32 @@ private function completeDefinition($id, Definition $definition)
             $this->container->addResource(static::createResourceForClass($reflectionClass));
         }
 
-        if ($constructor = $reflectionClass->getConstructor()) {
+        $constructor = $reflectionClass->getConstructor();
+        if ($constructor && (true === $autowired || '*' === $autowired)) {
             $this->autowireMethod($id, $definition, $constructor, true);
         }
 
+        if (true === $autowired) {
+            return;
+        }
+
         $methodsCalled = array();
         foreach ($definition->getMethodCalls() as $methodCall) {
             $methodsCalled[$methodCall[0]] = true;
         }
 
-        foreach (self::getSetters($reflectionClass) as $reflectionMethod) {
+        if (is_array($autowired)) {
+            $methods = array();
+            foreach ($autowired as $methodName) {
+                if ($reflectionClass->hasMethod($methodName)) {
+                    $methods[] = $reflectionClass->getMethod($methodName);
```

Comment:
Sure but it will add its arguments as a call instead of arguments. And it may create weird bugs and not work if there are mandatory arguments.